docsource/ops.rst

```diff
@@ -1,6 +1,236 @@
+          "properties": {
+            "extpub": {
+            "type": "integer"
+            },
+            . . .
+          }
+        }
+      },
+      "settings": {
+        "index": {
+          "number_of_shards": "5",
+          "number_of_replicas": "1"
+        }
+      }
+    }
+
+5c. If the mapping does not require any change but the documents' field values (not including meta-data fields whose key names start with underscore) need to be changed, use the `Painless <https://www.elastic.co/guide/en/elasticsearch/reference/5.4/modules-scripting-painless-syntax.html>`_ script to modify metadata. Setting version_type to external will cause Elasticsearch to preserve the version from the source index, create any documents that are missing, and update any documents that have an older version in the destination index than they do in the source index.
+
+.. code-block:: bash
+
+    POST _reindex
+    {
+      "source": {
+        "index": "kbase.genome_1"
+      },
+      "dest": {
+        "index": "kbase.genome_1_r1",
+        "version_type": "external"
+      },
+      "script": {
+        "lang": "painless",
+        "inline": "if (ctx._source.foo == 'bar') {ctx._version++; ctx._source.remove('foo')}"
+      }
+    }
+
+6. Now, reindex the entire data from current index to new index. Alternately, use a query to reindex only a subset of the current index.
+
+.. code-block:: bash
+
+    POST _reindex
+    {
+      "source": {
+        "index": "kbase.genome_1"
+      },
+      "dest": {
+        "index": "kbase.genome_1_r1"
+      }
+    }
+
+        OR
+
+    POST _reindex
+    {
+      "source": {
+        "index": "kbase.genome_1",
+        "query": {
+          ...
+        }
+      },
+      "dest": {
+        "index": "kbase.genome_1_r1"
+      }
+    }
+
+7. Run a checksum on the new index to make sure the numbers line up with the numbers of the current index.
+
+.. code-block:: bash
+
+    GET /kbase.genome_1_r1/_stats/docs,store
+
+8. Run a query to specifically check the change that was applied.
+
+.. code-block:: bash
+
+    GET kbase.genome_1_r1/_search
+
+   OR
+
+    GET kbase.genome_1_r1/_search
+    {
+     "query": {
+       "match": {
+         "FIELD": "VALUE"
+       }
+     }
+    }
+
+   OR
+
+    https://www.elastic.co/guide/en/elasticsearch/reference/5.5/search-request-body.html
+
+9. If the new index looks good, update index alias and delete current index.
+
+.. note::
+
+    If you want the current index to linger for a day or two to serve a rollback option, reindex the current index into another new index called kbase.genome_1_backup and then delete the current index. This is one of two ways of renaming an index in ElasticSearch. The other way is to use the snapshot API.
+
+.. code-block:: bash
+
+    POST _aliases
+    {
+     "actions": [
+     {
+       "add": {
+         "index": "kbase.genome_1_r1",
+         "alias": "kbase.genome"
+         }
+       },
+       {
+         "remove": {
```

Comment:
This is just removing the index alias, not the index itself, right? So it's just flipping the target of the alias?

If so this is necessary because you can't write to an alias that points to more than one index IIRC. Also, presumably you don't want to search both the old and the new index.